### ISSUE TITLE
Path handling issue for virtual files in Windows

### ISSUE THREAD

**Issue by octref** [vuejs/vetur#1235]
Forking off #1233:

@trixnz

Is this related to the issues with `sourceMap[filename]` being invalid? I've noticed on *Windows* that `getFilePath` and `getfileFsPath` are being used interchangeably to compare filepaths as strings, rather than  doing a proper comparison, honouring the slash differences on Windows.

These seem to propagate out to the `findReferences`, `findDefinition`, and `doHover` providers. After playing around with the comparisons, I managed to get it to properly provide those helpers, but it seems the file URIs returned must also be of the `file://` schema, otherwise VSCode doesn't recognise them (maybe another Windows specific requirement?)

If this is unrelated, let me know and i'll prepare another issue 😃. Love the work on interpolation, I can see this being awesome once the cross-platform issues are ironed out!

---

I pushed my WIP changes to https://github.com/trixnz/vetur/tree/wip-interpolation-compat, granted I didn't really know what I was doing as I'm still fairly new to the vetur codebase, but this got hover and go to definition working fine, but find all references still produces the wrong URI in the returned results, so you end up with the origin of the symbol being under the wrong path, but the template references are fine.

**Comment by trixnz** [comment]
I've updated my [branch](https://github.com/trixnz/vetur/commit/b06f738d629eaadde96f0379aa4b9a8a35ddfe5e) which seems to get hovering, find all references, and go to definition working correctly. It also fixes the show virtual file map, as it was also dependent on the source maps.

It adds `getNormalizedFilePathAsFileSchema` and `getNormalizedFilePath`, the latter of which can probably be merged with the existing path utils, although that has harder hitting effects, so I wasn't comfortable changing that.

It seems like Typescript returns its paths with forward slashes, but I don't know if that's consistent. File paths should probably always be normalised when doing comparisons except for when dealing with the filesystem.

I noticed `file://` (originating from vscode `TextDocument`) and absolute paths also seem to be passed around interchangably, which `getNormalizedFilePath` is capable of identifying and normalising to `/` slashes with absolute paths.
